pyiron/dft/job/generic.py

```diff
@@ -273,3 +267,42 @@ def set_empty_states(self, n_empty_states=None):
         raise NotImplementedError(
             "The set_empty_states function is not implemented for this code."
         )
+
+def get_k_mesh_by_cell(cell, kpoints_per_angstrom=1):
+    """
+    Args:
+        cell:
+        kpoints_per_angstrom:
+    Returns:
+    """
+    omega = np.linalg.det(cell)
+    l1, l2, l3 = cell
+    g1 = 2 * np.pi / omega * np.cross(l2, l3)
+    g2 = 2 * np.pi / omega * np.cross(l3, l1)
+    g3 = 2 * np.pi / omega * np.cross(l1, l2)
```

Comment:
Here's my problem: I do understand that 2pi has a physical origin, but in reality the k-mesh is solely determined empirically, i.e. people remember the numbers like 10, 12, 24 etc. I find it therefore confusing that there's a factor 2pi.

Reply:
I use 2*pi factor to be consistent with my previous "intuition" from  Materials Design's MedeA  software, where they take into account factor  2*pi when kmesh was setup from the density.
So, the 0.5 is very rough setting, whereas 0.15 or 0.1 is tight "enough"

Reply:
Ok at MPIE we usually use integers, but if there's another kind of background, I give in.